challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.map { |string| string.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.select { |string| string == "dog" }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.select { |string| string != "dog" }
```

Comment:
There's also a `reject` method that does the opposite of `select`. But, "select not dog" is plenty clear.